src/main/java/org/reactome/addlinks/referencecreators/HPAReferenceCreator.java

```diff
@@ -0,0 +1,72 @@
+package org.reactome.addlinks.referencecreators;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.List;
+import java.util.Map;
+
+public class HPAReferenceCreator extends SimpleReferenceCreator<List<String>>
```

Comment:
Some of the other new ReferenceCreators look pretty much identical to this one. This strongly suggests that some refactoring might be possible...

Reply:
Ya, as you saw above there was a number where I barely had to change anything, but they got there own class. Since there is so many 'samey' files being added, I'll look into rolling them together

Reply:
I'm not sure smushing them all into one class would be a good idea, but there might be some common functionality that can be pulled up in the parent class (maybe as a new method) and then call that from the child classes. Or it's possible that it all of the common functionality can be moved  up into `SimpleReferenceCreator` then you might not even need these classes. Just configure them as ReferenceCreators of type `SimpleReferenceCreator`, with the correct parameter values.

Reply:
I looked into doing that (all using SimpleReferenceCreator), but there isn't any current way of specifying which columns of the file to use. I could look into moving some samey methods to the SimpleReferenceCreator though which will cut down on a lot of the code

Reply:
It's hard to make a truly generic FileProcessor in AddLinks (so don't worry about it!), but for ReferenceCreators, it should be a bit easier.